agw/AUI.py demo asserts

When run the awg/AUI.py demo, it asserts, which looks like the same issue mentioned in http://trac.wxwidgets.org/ticket/17080

Apply the similar patch in http://trac.wxwidgets.org/attachment/ticket/17080/auibar.cpp.patch to wx/lib/agw/aui/auibar.py seems solve the issue.
 
> Traceback (most recent call last):
>   File "agw/AUI.py", line 3038, in OnButton1
>     self.win = MainAUI(self, self.log)
>   File "agw/AUI.py", line 3009, in MainAUI
>     frame = AuiFrame(parent, -1, "AUI Test Frame", size=(800, 600), log=log)
>   File "agw/AUI.py", line 910, in __init__
>     self.BuildPanes()
>   File "agw/AUI.py", line 1301, in BuildPanes
>     ToolbarPane().Top().Row(2))
>   File "/home/tianzhu/temp/Phoenix/wx/lib/agw/aui/framemanager.py", line 4735, in AddPane
>     return self.AddPane1(window, arg1)
>   File "/home/tianzhu/temp/Phoenix/wx/lib/agw/aui/framemanager.py", line 4804, in AddPane1
>     tb.SetGripperVisible(True)
>   File "/home/tianzhu/temp/Phoenix/wx/lib/agw/aui/auibar.py", line 2511, in SetGripperVisible
>     self.Realize()
>   File "/home/tianzhu/temp/Phoenix/wx/lib/agw/aui/auibar.py", line 3082, in Realize
>     ctrl_sizer_item = vert_sizer.Add(item.window, 0, wx.EXPAND)
> wx._core.wxAssertionError: C++ assertion ""!m_containingSizer"" failed at /home/tianzhu/temp/Phoenix/ext/wxWidgets/src/common/wincmn.cpp(2476) in SetContainingSizer(): Adding a window already in a sizer, detach it first!

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -97,6 +97,8 @@ Changes in this release include the following:
 
 * Use BufferedDC in Repaint() to mitigate flicker (wx.lib.agw.aui). (#686)
 
+* Detach wxControl in AuiToolbar from current sizer before attach to a new one.
+  (#843)
 
 4.0.1 "Lemonade"
 ----------------

--- a/wx/lib/agw/aui/auibar.py
+++ b/wx/lib/agw/aui/auibar.py
@@ -3072,6 +3072,9 @@ class AuiToolBar(wx.Control):
                     sizer_item = sizer.Add((item.spacer_pixels, 1))
 
             elif kind == ITEM_CONTROL:
+                if item.window and item.window.GetContainingSizer():
+                    # Make sure that there is only one sizer to this control
+                    item.window.GetContainingSizer().Detach(item.window);
 
                 vert_sizer = wx.BoxSizer(wx.VERTICAL)
                 vert_sizer.AddStretchSpacer(1)

